### PR TITLE
MULE-17770: Allow UTF-8 in disposition content headers for http request.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

**Review by eze210 — APPROVED** [review]

File: src/main/java/com/ning/http/client/multipart/PartBase.java [hunk 1]
```diff
@@ -89,11 +95,11 @@ protected void visitStart(PartVisitor visitor, byte[] boundary) throws IOExcepti
     protected void visitDispositionHeader(PartVisitor visitor) throws IOException {
         visitor.withBytes(CRLF_BYTES);
         visitor.withBytes(CONTENT_DISPOSITION_BYTES);
-        visitor.withBytes(getDispositionType() != null ? getDispositionType().getBytes(US_ASCII) : FORM_DATA_DISPOSITION_TYPE_BYTES);
+        visitor.withBytes(getDispositionType() != null ? getDispositionType().getBytes(getHeadersCharset()) : FORM_DATA_DISPOSITION_TYPE_BYTES);
```
Comment (afelisatti) [inline src/main/java/com/ning/http/client/multipart/PartBase.java]: It seems a lot of overhead to be asking this all the time when the property is defined statically.

File: src/main/java/com/ning/http/client/multipart/PartBase.java [hunk 2]
```diff
@@ -13,17 +13,23 @@
...
 import java.io.OutputStream;
 import java.nio.charset.Charset;
 import java.util.ArrayList;
 import java.util.List;
 
+import com.ning.http.client.Param;
+
 public abstract class PartBase implements Part {
+    
+    public static final String GRIZZLY_ALLOW_UTF8 = "grizzly.allowUtf8";
```
Comment (afelisatti) [inline src/main/java/com/ning/http/client/multipart/PartBase.java]: This property should be more specific and indicate it affects part handling.

File: src/test/java/com/ning/http/client/async/grizzly/GrizzlySimpleAsyncHttpClientAllowingUTF8Test.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import com.ning.http.client.AsyncHttpClientConfig;
+import com.ning.http.client.async.ProviderUtil;
+import com.ning.http.client.async.SimpleAsyncHttpClientTest;
+import com.ning.http.client.providers.grizzly.GrizzlyAsyncHttpProvider;
+
+public class GrizzlySimpleAsyncHttpClientAllowingUTF8Test extends SimpleAsyncHttpClientTest {
+
+    @BeforeClass
+    public void beforeClass() {
+      System.setProperty(GRIZZLY_ALLOW_UTF8, "true");
```
Comment (afelisatti) [inline src/test/java/com/ning/http/client/async/grizzly/GrizzlySimpleAsyncHttpClientAllowingUTF8Test.java]: As the property is static won't this mean the first load sticks?

**Review by afelisatti — CHANGES_REQUESTED** [review]

File: src/test/java/com/ning/http/client/async/grizzly/GrizzlySimpleAsyncHttpClientAllowingUTF8Test.java (hunk 3 above)
Comment (fsgonz) [reply src/test/java/com/ning/http/client/async/grizzly/GrizzlySimpleAsyncHttpClientAllowingUTF8Test.java]: Yes, it performs every test in SimpleAsyncHttpClientTest obtaining the string for each header as UTF-8. GrizzlySimpleAsyncHttpClient performs the same tests without the system property. GrizzlySimpleAsyncHttpClientAllowingUTF8Test also runs the tests where every character is ASCII valid.

File: src/main/java/com/ning/http/client/multipart/PartBase.java [hunk 4]
```diff
@@ -13,17 +13,23 @@
...
 import java.io.OutputStream;
 import java.nio.charset.Charset;
 import java.util.ArrayList;
 import java.util.List;
 
+import com.ning.http.client.Param;
+
 public abstract class PartBase implements Part {
+    
+    public static final String GRIZZLY_ALLOW_UTF8 = "grizzly.request.part.headers.allowUtf8";
```
Comment (afelisatti) [inline src/main/java/com/ning/http/client/multipart/PartBase.java]: `grizzly` -> `ahc`?

**Review by afelisatti — CHANGES_REQUESTED** [review]

File: src/test/java/com/ning/http/client/async/grizzly/GrizzlySimpleAsyncHttpClientAllowingUTF8Test.java (hunk 3 above)
Comment (fsgonz) [reply src/test/java/com/ning/http/client/async/grizzly/GrizzlySimpleAsyncHttpClientAllowingUTF8Test.java]: Ignore previous comment. Changed to use Reflection.

**Review by afelisatti — APPROVED** [review]

**Pull request merged**